Source/JavaScriptCore/wasm/js/WebAssemblyFunction.cpp

```diff
@@ -83,7 +84,11 @@ JSC_DEFINE_HOST_FUNCTION(callWebAssemblyFunction, (JSGlobalObject* globalObject,
         return throwVMTypeError(globalObject, scope, Wasm::errorMessageForExceptionType(Wasm::ExceptionType::TypeErrorInvalidV128Use));
 
     for (unsigned argIndex = 0; argIndex < signature.argumentCount(); ++argIndex) {
-        uint64_t value = fromJSValue(globalObject, signature.argumentType(argIndex), callFrame->argument(argIndex));
+        auto argType = signature.argumentType(argIndex);
+        uint64_t value = fromJSValue(globalObject, argType, callFrame->argument(argIndex));
+        if (UNLIKELY(argType.isRef()))
```

Comment:
So, the reason for checking `Ref` is here is that it's introduced by WASM GC proposal, which can represents the complex data structures. What about the other types? Are they not managed by WASM GC?

Reply:
This is only needed if a new value is produced by this wrapper; If we are just copying from the stack, then it is already kept alive